packages/app/src/cli/services/context.ts

```diff
@@ -430,7 +430,7 @@ export async function ensureDeployContext(options: DeployContextOptions): Promis
   const {reset, force, noRelease} = options
   let developerPlatformClient = options.developerPlatformClient
   // do the link here
-  const [remoteApp] = await fetchAppAndIdentifiers(options, developerPlatformClient, true, true)
+  const [remoteApp] = await fetchAppAndIdentifiers(options, developerPlatformClient, true, !options.apiKey)
```

Comment:
Positional arguments here makes it hard to see, so here's a link to the API we're calling to demonstrate that the `enableLinkingPrompt` is what changed: https://github.com/Shopify/cli/blob/18fae61a9fc8a1866e0dc80d4219fd592ece170f/packages/app/src/cli/services/context.ts#L645